vaccine/states.py

```diff
@@ -254,3 +254,32 @@ async def display(self, message: Message):
                 ],
             },
         )
+
+
+class SectionedChoiceState(ChoiceState):
+    def __init__(self, *args, sections: dict, separator: Optional[str] = "", **kwargs):
+        choices = []
+        for section_choices in sections.values():
+            choices.extend(section_choices)
+
+        kwargs["choices"] = choices
+
+        super().__init__(*args, **kwargs)
+        self.sections = sections
+        self.separator = separator
+
+    @property
+    def _display_choices(self) -> str:
+        lines = []
+
+        i = 1
+        for section, choices in self.sections.items():
+            lines.append(f"*{section}*")
+
+            for choice in choices:
+                lines.append(f"{i}. {choice.label}")
+                i += 1
+
+            lines.append(self.separator or "")
```

Comment:
Since `separator` is `Optional`, maybe we should take advantage of that, and leave the separator out if it's `None`. The way it is now, you can't not have a newline between sections.
ie.
```
if self.separator is not None:
    lines.append(self.separator)
```